src/modules/SourceDialogs.rb

```diff
@@ -375,12 +375,19 @@ def IsISOURL(url)
     def PreprocessISOURL(url)
       log.info "Preprocessing ISO URL: #{URL.HidePassword(url)}"
 
-      uri = (url == "iso://") ? URI("iso:/") : URI(url)
+      # empty iso is used when we adding new url
+      return "" if url == "iso://"
+      uri = URI(url)
       query = uri.query || ""
-      params = URI.decode_www_form(query).to_h
 
-      processed = URI(params.delete("url") || "")
-      processed.path = File.join(processed.path || "", params.delete("iso") || "")
+      # libzypp do not use web encoding as in https://www.w3.org/TR/html5/forms.html#url-encoded-form-data
+      # but percentage enconding only. For more details see (bsc#954813#c20)
+      params = URI.decode_www_form(query.gsub(/%20/, "+")).to_h
+
+      param_url = params.delete("url") || ""
+      processed = URI(URI.encode(param_url))
+      processed.scheme = "iso" if processed.scheme.downcase == "dir"
```

Comment:
What about the other scheme like `file`, we do not need to handle them? Please add a comment if needed...